packages/mui-material/src/ListItem/ListItem.spec.tsx

```diff
@@ -1,35 +1,16 @@
 import * as React from 'react';
 import ListItem from '@mui/material/ListItem';
 import { styled } from '@mui/material/styles';
-
-// button: boolean
-function BooleanButtonTest() {
-  // https://github.com/mui/material-ui/issues/14971
-
-  function EditableItemFail(props: { editable: boolean }) {
-    const { editable } = props;
-    // @ts-expect-error 'boolean' is not assignable to type 'true'
-    return <ListItem button={editable}>Editable? {editable}</ListItem>;
-  }
-
-  function EditableItemValid(props: { editable: boolean }) {
-    const { editable } = props;
-    if (editable) {
-      <ListItem button>Editable? Yes</ListItem>;
-    }
-    return <ListItem>Editable? No</ListItem>;
-  }
-}
+import ListItemButton from '@mui/material/ListItemButton';
 
 // verify that https://github.com/mui/material-ui/issues/19756 already worked.
 function MouseEnterTest() {
   function handleMouseEnter(event: React.MouseEvent<HTMLLIElement>) {}
   <ListItem onMouseEnter={handleMouseEnter} />;
 
   function handleMouseEnterButton(event: React.MouseEvent<HTMLDivElement>) {}
-  // @ts-expect-error
   <ListItem onMouseEnter={handleMouseEnterButton} />; // desired: missing property button
```

Comment:
How was the ts error implemented before? We might want to remove this case completely but first I want to understand why it was expected to error before.

Reply:
The error was the same as I mentioned in the previous comments: Due to the dependency NOT being removed, the event on Line 11 is of `HTMLDivElement` which was expected. So the `@ts-expect-error` could be removed. But on removing the dependency, it is actually of type `HTMLLIElement`, so the directive is needed to be present, because the `MouseEvent` expects `HTMLDivElement`, but it receives `HTMLLIElement`. I have reverted this line as well.